source/Octopus.Server.Client/Repositories/Async/BasicRepository.cs

```diff
@@ -319,6 +319,44 @@ public async Task<List<TResource>> FindByNames(IEnumerable<string> names, string
                 return false;
             }, path, pathParameters, cancellationToken).ConfigureAwait(false);
         }
+        
+        public Task<TResource> FindBySlug(string slug, CancellationToken cancellationToken)
+        {
+            return FindBySlug(slug, null, null, cancellationToken);
+        }
+        
+        public async Task<TResource> FindBySlug(string slug, string path, object pathParameters, CancellationToken cancellationToken)
+        {
+            await ThrowIfServerVersionIsNotCompatible(cancellationToken).ConfigureAwait(false);
+
+            slug = (slug ?? string.Empty).Trim();
+
+            // Some endpoints allow a Slug query param which greatly increases efficiency
+            pathParameters ??= new { slug };
```

Comment:
I'm not sure if there are currently any endpoints that have a slug parameter, but this isn't an issue to include